### ISSUE TITLE
Checkbox focus outline not visible in high contrast mode

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#6471]
## Environment

> OS

Windows 10 etc.

> Browsers

IE11, Edge Chromium, Firefox

> Settings

High Contrast Theme #1 enabled

## Detailed description

> What version of the Carbon Design System are you using?

10.14

> What did you expect to happen?

To see a focus outline on the checkbox, like when high contrast mode is disabled.

> What happened instead?

No focus outline.

> What WCAG 2.1 checkpoint does the issue violate?

https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

## Steps to reproduce the issue

The problem is not specific to react but you can see it at https://react.carbondesignsystem.com/?path=/story/checkbox--checked.

## Additional information

Here's what it looks like, regardless of whether or not it's focused:

<img width="173" alt="Screen Shot 2020-07-15 at 15 32 26" src="https://user-images.githubusercontent.com/69599/87511704-b9161f80-c6b0-11ea-9f5d-42bb9af80c8d.png">

The problem  is that the focus outline is rendered via `box-shadow` rather than outline:

```
  // Unchecked
  .#{$prefix}--checkbox:focus + .#{$prefix}--checkbox-label::before,
  .#{$prefix}--checkbox-label__focus::before,
  // Checked
  .#{$prefix}--checkbox:checked:focus + .#{$prefix}--checkbox-label::before,
  .#{$prefix}--checkbox-label[data-contained-checkbox-state='true'].#{$prefix}--checkbox-label__focus::before,
  // Indeterminate
  .#{$prefix}--checkbox:indeterminate:focus + .#{$prefix}--checkbox-label::before,
  .#{$prefix}--checkbox-label[data-contained-checkbox-state='mixed'].#{$prefix}--checkbox-label__focus::before {
    // Must use box-shadow for appearance of multiple borders with rounded corners.
    box-shadow: 0 0 0 2px $inverse-01, 0 0 0 4px $focus;
  }
```

The obvious fix is to use `outline` instead of `box-shadow` but assuming there's some reason you aren't doing that already, a possible alternative fix is:

```
@media screen and (-ms-high-contrast: active) {
	// Unchecked
	.#{$prefix}--checkbox:focus + .#{$prefix}--checkbox-label::before,
	.#{$prefix}--checkbox-label__focus::before,
	// Checked
	.#{$prefix}--checkbox:checked:focus + .#{$prefix}--checkbox-label::before,
	.#{$prefix}--checkbox-label[data-contained-checkbox-state='true'].#{$prefix}--checkbox-label__focus::before,
	// Indeterminate
	.#{$prefix}--checkbox:indeterminate:focus + .#{$prefix}--checkbox-label::before,
	.#{$prefix}--checkbox-label[data-contained-checkbox-state='mixed'].#{$prefix}--checkbox-label__focus::before {
		outline: 1px solid highlightText;
		outline-offset: 2px;
	}
}
```

**Comment by jendowns** [comment]
@carbon-design-system/ibma Can the focus state for the checkboxes be revisited... I have noticed positioning issues (see #6480), and then just saw this more serious issue about the focus state not being visible in high contrast mode.

The use of box-shadow for the checkbox's focus state was originally to help show multiple borders with "rounded corners". But can this library move away from that and return to using outlines (or a single border) -- for accessibility, consistency, and stability?

**Comment by dakahn** [comment]
@jendowns I'm absolutely on board with this change.

**Comment by wkeese** [comment]
BTW, I noticed that `box-shadow` is used for buttons too.  Actually a `border` + `box-shadow` combo.  Seems like carbon really doesn't like `outline`.

**Comment by tw15egan** [comment]
@carbon-design-system/design any chance we can take a look at revisiting Checkbox focus states?

**Comment by aagonzales** [comment]
I'm not sure what else we could visually change the focus state to. Most focus styles come in some form of a border. Is a style of a border not possible here? What are some dev solutions that would resolve this issue?

**Comment by laurenmrice** [comment]
Did the update we did recently to the focus states help this? The focus border now has a gap between the border and the icon.

**Comment by tw15egan** [comment]
I think we were originally using `box-shadow` to show multiple borders with rounded corners, not sure if that has changed. Possible solutions would be to use an `outline` or `border`

**Comment by wkeese** [comment]
https://react.carbondesignsystem.com/?path=/story/checkbox--checkbox is still using the box shadow, so still not working (unless you are talking about a change not released yet).

**Comment by tw15egan** [comment]
Working on a HCM fix, but if we want to revisit the focus states lets open up a separate issue for that 👍

**Comment by wkeese** [comment]
@tw15egan, can you reopen this?  It doesn't work in Edge or Firefox.  Here's an example of a focused checkbox and an unfocused one (on Edge), from https://react.carbondesignsystem.com/?path=/story/checkbox--checkbox:

<img width="194" alt="Screen Shot 2021-01-14 at 10 06 42" src="https://user-images.githubusercontent.com/69599/104530936-63ff9a00-5650-11eb-8f34-30094b5b2110.png">

**Comment by tw15egan** [comment]
Power is out at my house right now, but once it's back on I'll boot up my Windows machine and investigate what's going on with these...
